src/index/txospenderindex.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <common/args.h>
+#include <index/txospenderindex.h>
+#include <logging.h>
+#include <node/blockstorage.h>
+#include <validation.h>
+
+// LeveLDB key prefix. We only have one key for now but it will make it easier to add others if needed.
+constexpr uint8_t DB_TXOSPENDERINDEX{'s'};
+
+std::unique_ptr<TxoSpenderIndex> g_txospenderindex;
+
+/** Access to the txo spender index database (indexes/txospenderindex/) */
+class TxoSpenderIndex::DB : public BaseIndex::DB
+{
+public:
+    explicit DB(size_t n_cache_size, bool f_memory = false, bool f_wipe = false);
+
+    bool WriteSpenderInfos(const std::vector<std::pair<COutPoint, uint256>>& items);
+    bool EraseSpenderInfos(const std::vector<COutPoint>& items);
+};
+
+TxoSpenderIndex::DB::DB(size_t n_cache_size, bool f_memory, bool f_wipe) : BaseIndex::DB(gArgs.GetDataDirNet() / "indexes" / "txospenderindex", n_cache_size, f_memory, f_wipe)
```

Comment:
nit: I like how you responded to https://github.com/bitcoin/bitcoin/pull/24539#discussion_r1704599014 but my primary intention was actually doing:
```suggestion
TxoSpenderIndex::DB::DB(size_t n_cache_size, bool f_memory, bool f_wipe)
    : BaseIndex::DB(gArgs.GetDataDirNet() / "indexes" / "txospenderindex", n_cache_size, f_memory, f_wipe)
```